### DISCUSSION TITLE
[Feature Request] Add option to bypass TPM 2.0, Secure Boot and remove MS Online Account in Win 11

### DISCUSSION THREAD

**Discussion by Dark-Cod3r** [ventoy/Ventoy#1861] [Ideas]
Hi. Rufus added options for windows 11 such as bypass tpm 2.0, bypass secure boot and remove Microsoft online account requirement. If possible, add these options in Ventoy as well.
Thank You
[Check This Image](https://i.postimg.cc/C5WvDbpS/capture-09202022-210004.jpg)

**Comment by ventoy** [comment]
There is an `VTOY_WIN11_BYPASS_CHECK` option in GlobalControl plugin.
https://www.ventoy.net/en/plugin_control.html

**Comment by ventoy** [comment]
As for the online account, refer: #1662

**Comment by Dark-Cod3r** [comment]
Thank You.. Will look into it..

**Comment by ruftas** [comment]
I learned something new here

**Comment by Korb** [comment]
> Rufus added options for windows 11 such as bypass tpm 2.0, bypass secure boot and remove Microsoft online account requirement. If possible, add these options in Ventoy as well.

Here's a full list of Rufus's features, which make it quicker for me to create a custom, modified bootable USB drive for Windows 11 than to configure a JSON file:

> **Windows User Experience**
> Customize Windows installation?
> 
> 1. Remove requirement for 4GB+ RAM, Secure Boot and TPM 2.0
> 2. Remove requirement for an online Microsoft account
> 3. Create a local account with username
> 4. Set regional options to the same values as this user's
> 5. Disable data collection (Skip privacy questions)
> 6. Disable BitLocker automatic device encryption

Yes, it's inconvenient to have multiple bootable USB drives. This is something that Ventoy successfully combats in most cases. But not in this particular case. Unfortunately. The very possibility of doing something (even more flexibly) sometimes does not outweigh the inconvenience.

**Comment by Tim0219800** [comment]
**Titre :** VentoyStudio — a GUI to generate autounattend.xml and ventoy.json for
Windows images

---

Ventoy already has everything needed to apply an answer file: `auto_install`,
several templates per image, a boot menu to pick between them, variable
expansion. What it has never had is something to *write* those files.

So I built one: **VentoyStudio**, a small Windows GUI that generates the
`autounattend.xml` and merges the matching `auto_install` entry into
`ventoy/ventoy.json`.

https://github.com/Tim0219800/VentoyStudio

**What it does**

- reads the ISO without mounting it — version, build, editions, language (UDF
  first, ISO9660 as a fallback)
- local account, no mandatory Microsoft account, OOBE fully pre-answered
- debloat, dark theme, wallpaper, taskbar settings
- 71 winget packages in a searchable catalogue, or any id you type
- a custom PowerShell script at first logon
- **several profiles per image**, with `autosel`, `timeout` and the
  `auto_install_uefi` / `_legacy` suffixes — the part no other tool exposes
- installs Ventoy itself on a blank drive, or removes it

**What it deliberately does not do**

It never modifies Ventoy, GRUB or your ISOs. It never writes
`DiskConfiguration`, so the disk and partitions stay the user's choice in Setup.
It writes no `LabConfig` keys either, since Ventoy already applies the hardware
bypass earlier and more cleanly than an answer file can — the UI only exposes
the ability to turn that default off.

`ventoy.json` is **merged, never regenerated**: plugins it does not manage
survive byte for byte.

**Status**

Validated end to end by a full Windows 11 25H2 install in a VM — from the drive
to the desktop, no questions asked, software installed. 157 frontend and 67
backend tests. MIT licensed. Ventoy is not bundled: it is downloaded from its
official release and its SHA-256 verified before anything runs.

The binaries are unsigned, so Windows will show a SmartScreen warning. SHA-256
sums are in the release notes.

Feedback welcome, especially on the `auto_install` edge cases — a lot of the
behaviour I relied on is only visible in `ventoy_plugin.c`, and I would rather
be told now if I read something wrong.
